### PR TITLE
Clean up design docs

### PR DESCRIPTION
**Pull request by xylar**
This merge clean up the existing design docs so they integrate more cleanly into the documentation as a whole.  This includes:
* giving each design doc an anchor (like `(omega-design-logging)=`) in case we want to link to it in the docs
* removing the `OMEGA Requirements and Design:` super-title, because it shows up as the document name in the table of contents
* Explicitly formatting code as c++, yaml, etc. for improved readability

Checklist
* [x] Documentation:
  * [x] Design document has been updated
  * [x] Documentation has been [built locally](https://e3sm-project.github.io/omega/develop/developers_guide/building_docs.html) and changes look as expected

### CODE REVIEW

**Comment by xylar** [comment]
Builds on #14 so I will rebase after that gets merged

**Review by grnydawn — APPROVED** [review]
Overall, the html pages from Sphinx-build looks good to me.

One suggestion for possibly for future is about using cross-referencing. So far, we have a simple document structure with the three top-level navigation menu, and I guess the # of menus and submenus would grow so that sometime it is hard to locate the page I wish to refer during reading one page in the document. In the case, cross-reference link at the word/phrase of the page might be helpful to easily navigate.

**Comment by xylar** [comment]
@grnydawn, thanks for reviewing! 

> One suggestion for possibly for future is about using cross-referencing. So far, we have a simple document structure with the three top-level navigation menu, and I guess the # of menus and submenus would grow so that sometime it is hard to locate the page I wish to refer during reading one page in the document. In the case, cross-reference link at the word/phrase of the page might be helpful to easily navigate.

Yes, that's a great suggestion and it's consistent with my plan once we start to flesh out the user's and developer's guides.  In my view, it's probably good enough to just have the one link to the top of each design document because we won't typically reference them elsewhere in the documentation at all, but is is very easy to add anchors wherever they are useful within a MyST document and to reference them elsewhere in the documentation.

As we get a bit further along, I'll work with all of you to get the early documentation in place, and hopefully, that will provide a good example for additional documentation down the road.

Please keep the suggestions coming!

File: components/omega/doc/devGuide/BuildDocs.md
Comment (xylar) [inline components/omega/doc/devGuide/BuildDocs.md:5]: @grnydawn, I've added some more documentation on the documentation itself to #14 (included here as well). Here is an example of how you can internally link to another section within the documentation by referencing its anchor.  Let me know if this is the type of link you were thinking of in your suggestion.

**Review by philipwjones — APPROVED** [review]
The modified design docs look good, at least in the Github viewer. Thanks for translating them all.

**Comment by xylar** [comment]
@brian-oneill, could you take a look at the changes I made to the design doc(s) you have contributed and let me know if you're okay with them?

**Review by brian-oneill — APPROVED** [review]
Looks good, the explicit language code blocks are an improvement

**Comment by xylar** [comment]
Thanks @brian-oneill!  And no worries, I just marked this as ready for review this morning.

@philipwjones, would you be willing to merge this one, too when you get a chance?

**Pull request merged**
